tuxemon/core/states/dialog/__init__.py

```diff
@@ -60,3 +63,29 @@ def next_text(self):
             return text
         except IndexError:
             return None
+
+    def draw_avatar(self):
+        if not self.avatar:
+            return
+
+        # If the prefix is "monster" the second parameter refers to a monster
+        # If this parameter is a number, we're referring to a monster slot in the party
+        # If this parameter is a string, we're referring to a monster name
+        # If there's no prefix, the parameter represents the path to a sprite file
+        params = self.avatar.split(" ")
+        if params[0] == "monster":
+            if params[1].isdigit():
+                player = self.game.player1
+                slot = int(params[1])
+                avatar_sprite = player.monsters[slot].get_sprite("menu")
+            else:
+                avatar_monster = monster.Monster()
+                avatar_monster.load_from_db(params[1])
+                avatar_monster.flairs = {} # Don't use random flair graphics
+                avatar_sprite = avatar_monster.get_sprite("menu")
+        else:
+            avatar_sprite = tools.load_sprite(self.avatar)
```

Comment:
I think i would rather not load any avatar if the menu sprite is not available.  do you think it would look strange?  im not commited to this either way though

Reply:
It might indeed be a good idea to remove that. We'll need to support avatars for NPC's later down the road, but it should indeed be done using a similar syntax to monsters now.

Reply:
Fixed in the latest commit: The parameter now takes only one value, which is either a number or a string. When a number it's a monster slot, when a string it refers to a monster by name. Direct file paths can no longer be specified.